src/extension/src/ActionHandler.py

```diff
@@ -112,30 +112,43 @@ def tear_down(self):
             self.file_logger = None
 
     def setup_telemetry(self):
-        """ Init telemetry if agent is compatible (env var check) AND events_folder is specified.
+        """ Init telemetry if agent is compatible (events_folder is specified).
             Otherwise, error since guest agent does not support telemetry. """
         events_folder = self.ext_env_handler.events_folder
-        if events_folder is not None and self.telemetry_writer.is_agent_compatible():
+        if events_folder is not None:
             # Guest agent fully supports telemetry
-            ''' NOTE: unlike core, this code will run even if events_folder does not exist, 
-                since telemetry_writer.is_agent_compatible() only checks the env var. 
-                This ensures that the events_folder exists once core runs. '''
+            self.__log_telemetry_info(telemetry_supported=True)
+
             if not os.path.exists(events_folder):
                 os.mkdir(events_folder)
                 self.logger.log("Events folder path found in HandlerEnvironment but does not exist on disk. Creating now. [Path={0}]".format(str(events_folder)))
```

Comment:
Please log the agent version where this was observed here too so this type of bug in the agent is easily measurable at scale from a single match (vs. going over multiple per machine).

Reply:
Agent version is included in `__log_telemetry_info` - do you mean a separate log with just the agent version?

Reply:
At the end of this log line -- "Creating now. [Path={0}][AgentVersion={1}]"